rtsp.go

```diff
@@ -705,3 +697,74 @@ func getAvailableCodec(session *description.Session) videoCodec {
 
 	return Unknown
 }
+
+func (rc *rtspCamera) storeH264Frame(au [][]byte) {
+	naluIndex := 0
+	for naluIndex < len(au) {
+		nalu := au[naluIndex]
+		if isCompactableH264(nalu) {
+			// if the NALU is a compactable type, compact it, feed it into the decoder & skip
+			// the NALUs that were compacted.
+			// We do this so that the libav functions the decoder uses under the hood don't log
+			// spam error messages (which happens when it is fed SPS or PPS without an IDR
+			nalu, nalusCompacted := rc.compactH264SPSAndPPSAndIDR(au[naluIndex:])
+			if err := rc.decodeAndStore(nalu); err != nil {
+				rc.logger.Debugf("error decoding(2) h264 rtsp stream  %s", err.Error())
+				return
+			}
+			naluIndex += nalusCompacted
+			continue
+		}
+
+		// otherwise feed in each non compactable NALU into the decoder
+		if err := rc.decodeAndStore(nalu); err != nil {
+			rc.logger.Debugf("error decoding(2) h264 rtsp stream  %s", err.Error())
+			return
+		}
+		naluIndex++
+	}
+}
+
+func (rc *rtspCamera) compactH264SPSAndPPSAndIDR(au [][]byte) ([]byte, int) {
+	compactedNALU, numCompacted := []byte{}, 0
+	for _, nalu := range au {
+		if !isCompactableH264(nalu) {
+			// return once we hit a non SPS, PPS or IDR message
+			return compactedNALU, numCompacted
+		}
+		// If this is the first iteration, don't add the start code
+		// as the first nalu has not been written yet
+		if len(compactedNALU) > 0 {
+			startCode := H2645StartCode()
+			compactedNALU = append(compactedNALU, startCode...)
+		}
+		compactedNALU = append(compactedNALU, nalu...)
+		numCompacted++
+	}
+	return compactedNALU, numCompacted
+}
+
+// H2645StartCode is start code byte sequence for H264/H265 NALs.
+func H2645StartCode() []byte {
```

Comment:
possible [nit] - should we make this a const?

Reply:
I don't think so given that append can mutate the underlying slice, which means that  there is a risk that data from other messages could leak into the global